インクリメンタルサーチ　グループメンバー検索機能実装

shiratori-sh opened the pull request:
#WHAT
グループ編集ページ、グループ新規作成ページのインクリメンタルサーチ実装
Turbolinksイベントハンドラ登録定義し直し。
html追加要素をjavascript上で追加。
ブループメンバー追加
グループメンバー削除機能実装。

キャプチャgifリンク
https://gyazo.com/0c8b17cc4cd92500010ccc46e0f00a59

#WHY
動作高速化のため。
検索しやすくするため。

shiratori-sh committed:
インクルメンタルサーチ

グループメンバー検索機能実装。

shiratori-sh committed:
グループメンバー追加、削除機能実装

Turbolinksイベントハンドラ登録定義し直し。
html追加要素をjavascript上で追加。
ブループメンバー追加
グループメンバー削除機能実装。

hiroki5123 commented on app/assets/javascripts/add_user.js:
```diff
@@ -0,0 +1,74 @@
+    user_list.append(html);
+  }
+
+function appendUsersErrorHTML(error){
```
インデントずれています

shiratori-sh replied:
インデントを調整しました。

hiroki5123 commented on app/assets/javascripts/add_user.js:
```diff
@@ -0,0 +1,74 @@
+            })
+          }
+          else{
+            appendUsersErrorHTML("一致するユーザーは見つかりません");
```
この場合だと引数として渡す必要ないと思います。
メソッドの中に直接書いた方がいいんじゃないでしょうか？

shiratori-sh replied:
テンプレートリテラル記法のHTML内に直接記述しました。引数を消しました。

hiroki5123 commented on app/assets/javascripts/add_user.js:
```diff
@@ -0,0 +1,74 @@
+      }
+  });
+  var group_member = $("#chat-group-users");
+function append(member, member_id) {
```
インデント一個奥です

shiratori-sh replied:
インデント調整しました。

hiroki5123 commented on app/assets/javascripts/add_user.js:
```diff
@@ -0,0 +1,74 @@
+}
+
+  $('#user-search-result').on('click', '.user-search-add', function(){
+  $(this).parent().empty();
```
functionのなかはインデント奥にしてください

shiratori-sh replied:
インデント調整しました。

shiratori-sh committed:
指摘内容調整

インデント位置合わせ
グループメンバーの検索でのエラーメッセージを引数で渡していたものをテンプレートリテラル記法で直接記述。

hiroki5123 commented on app/assets/javascripts/add_user.js:
```diff
@@ -0,0 +1,78 @@
+      }
+  });
+  var group_member = $("#chat-group-users");
+  function append(member, member_id) {
```
このメソッド名がappendと被っていてわかりにくいので修正しましょう

shiratori-sh replied:
function append(member, member_id)
→function appendRemoveUser(member, member_id)
メソッド名を変更しました。

shiratori-sh replied:
すみません。エラーが出ているのを確認していいませんでした。再度確認します。

shiratori-sh replied:
function append(member, member_id)
→ function appendRemoveUser(member, member_id)
メソッド名修正後　修正忘れの同メソッド修正しました。
append(member, member_id)
→appendRemoveUser(member, member_id)

shiratori-sh committed:
インクルメンタルサーチ　メソッド名修正

function append(member, member_id)
→function appendRemoveUser(member, member_id)

shiratori-sh committed:
再　インクルメンタルサーチ　メソッド名修正

 function append(member, member_id)
→ function appendRemoveUser(member, member_id)
メソッド名修正後　修正忘れの同メソッド修正
   append(member, member_id)
→appendRemoveUser(member, member_id)

shiratori-sh commented:
ありがとうございます。マージします。

hiroki5123 commented on app/views/groups/_form.html.haml:
```diff
@@ -11,31 +11,33 @@
+          %p.chat-group-user__name
+            - @group.users.each do |user|
+              = "#{user.name},"
+  / %a.user-search-add.chat-group-user__btn.chat-group-user__btn--add{"data-user-id" => "${ user.user_id }", "data-user-name" => "${ user.user_name }"} 追加
```
不要であれば削除しましょう

hiroki5123 commented on app/views/groups/_form.html.haml:
```diff
@@ -11,31 +11,33 @@
+      .chat-group-form__search.clearfix
+        %input#user-search-field.chat-group-form__input{placeholder: "追加したいユーザー名を入力してください", type: "text"}/
+      #user-search-result
+      / <div class='chat-group-form__field--left'>
```
不要であれば削除してください
